de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/Wikipedia.java

```diff
@@ -73,8 +82,38 @@ public Wikipedia(DatabaseConfiguration dbConfig) throws WikiInitializationExcept
         this.idMapCategories = new HashMap<Integer,Long>();
 
         this.metaData = new MetaData(this);
+        this.wikiConfig = getWikiconfig(this.language);
 	}
 
+    /**
+     * Configures a language specific configuration for parsing wikipedia pages.
+     * @param lang language parameter using for configuration
+     * @return WikiConfig
+     */
+    public static WikiConfig getWikiconfig(Language lang) {
+        WikiConfig config = DefaultConfigEnWp.generate();
+        if (lang != Language._test) {
+            // We need to capitalize the language name otherwise the locale lib cannot find it.
+            String langName = lang.name().substring(0, 1).toUpperCase() + lang.name().substring(1);
+            try {
+                List<LanguageCode> langCodes = LanguageCode.findByName(langName);
+                if (!langCodes.isEmpty()) {
+                    String langCode = langCodes.get(0).name();
+                    return LanguageConfigGenerator.generateWikiConfig(langCode);
+                } else {
+                    System.out.println(String.format("Cannot find language codes for %s", langName));
```

Comment:
Doesn't JWPL use a logging backend yet?

Reply:
This was used to test this function. My idea is to move the function that generates the language code to the Language enum class and if there's a logging backend there, I will use that. Otherwise I will remove the print statement

Reply:
Apache Commons logging should always be usable. If somebody wants to contribute more time, we could move to SLF4J, but that should be handled in a different issue.